ideascube/mediacenter/tests/test_models.py

```diff
@@ -0,0 +1,38 @@
+import pytest
+
+from .factories import DocumentFactory
+
+from os.path import basename
+
+@pytest.mark.django_db(transaction=True)
+def test_deleting_document_should_delete_file_also(mocker):
+    file_name = 'foo.img'
+    exist_mock = mocker.patch('os.path.exists')
+    exist_mock.side_effect = lambda p: basename(p) == file_name
+    remove_mock = mocker.patch('os.remove')
+    document = DocumentFactory(original=file_name)
+    document.delete()
+    assert remove_mock.call_count == 1
+
+
+@pytest.mark.django_db(transaction=True)
+def test_deleting_document_should_delete_preview_also(mocker):
+    file_name = 'preview_foo.img'
+    exist_mock = mocker.patch('os.path.exists')
+    exist_mock.side_effect = lambda p: basename(p) == file_name
+    remove_mock = mocker.patch('os.remove')
+    document = DocumentFactory(preview=file_name)
+    document.delete()
+    assert remove_mock.call_count == 1
+
+
+@pytest.mark.django_db(transaction=True)
+def test_deleting_document_should_delete_file_and_preview_also(mocker):
+    file_name = 'oo.img'
+    preview_name = 'preview_foo.img'
+    exist_mock = mocker.patch('os.path.exists')
+    exist_mock.side_effect = lambda p: basename(p) in (file_name, preview_name)
+    remove_mock = mocker.patch('os.remove')
```

Comment:
> And in fact, did you test it?

Of course I've tested it.

I've checked the code of django's FileField.delete() and it is globaly:
```python
if os.path.exists(path):
    os.remove(path)
```
(https://github.com/django/django/blob/1.10.5/django/core/files/storage.py#L379)
So we have to mock both to be sure that, at end, `os.remove` is called.

However, checking on github, I've just seen that on `master` they use a more pythonist way with a `try/catch` (https://github.com/django/django/blob/master/django/core/files/storage.py#L379)

So we need to have a more complex mock of os.remove than just mocking with a magicmock and we need a second test to be sure that every thing is ok if the file doesn't exist.